.github/workflows/rabbitmq.yml

```
@@ -0,0 +1,68 @@
+name: rabbitmq
+
+on:
+    push:
+        branches-ignore: [gh-pages]
+    pull_request:
+        branches-ignore: [gh-pages]
+        paths-ignore: ['docs/**']
+
+jobs:
+
+    tests:
+
+        runs-on: ubuntu-latest
+        timeout-minutes: 30
+
+        strategy:
+            fail-fast: false
+            matrix:
+                rabbitmq: [3.5, 3.6, 3.7, 3.8]
+
+        services:
+            postgres:
+                image: postgres:10
+                env:
+                    POSTGRES_DB: test_django
+                    POSTGRES_PASSWORD: ''
+                    POSTGRES_HOST_AUTH_METHOD: trust
+                options: >-
+                    --health-cmd pg_isready
+                    --health-interval 10s
+                    --health-timeout 5s
+                    --health-retries 5
+                ports:
+                -   5432:5432
+            rabbitmq:
+                image: rabbitmq:${{ matrix.rabbitmq }}
+                ports:
+                -   5672:5672
```

```suggestion
                ports:
                -   5672:5672
                options: >-
                    --health-cmd rabbitmq-diagnostics -q ping
                    --health-interval 30s
                    --health-timeout 30s
                    --health-retries 5
```

this I believe is the recommended health check for rabbitmq (see `.ci/docker-rabbitmq.yml`) but obviously not strictly

---

it seems like adding these options has caused shit to stop working. Is the syntax different with GHA? It doesn't seem to like the `-q` flag. I think because of quoting probably, it should be interpreted as part of the value for `--health-cmd` but is seen as a separate flag.

---

yep that should do it

---

weird, I literally copy/pasted the command GH generates and it run fine locally: `docker create --name 9489d5a84fa44440b65da8d4368eb107_rabbitmq36_8515e2 --label 179394 --network github_network_bba19cb5634245c991e75611b37d8419 --network-alias rabbitmq -p 5672:5672 --health-cmd 'rabbitmq-diagnostics -q ping' --health-interval 30s --health-timeout 30s --health-retries 5 -e GITHUB_ACTIONS=true -e CI=true rabbitmq:3.6`

---

maybe double-quote or bust

---

> unknown shorthand flag: 'q' in -q

---

double quoting seems to have worked and it now runs for 3.7 and 3.8. However, for 3.5 and 3.6 the health checks fail. Are these options specific for the newer versions?